src/Options.dfy

```diff
@@ -0,0 +1,42 @@
+// RUN: %dafny "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+/*********************************************************************************************
+*  Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT 
+**********************************************************************************************/
+
+module Options {
```

Comment:
It feels weird to me to define `Result` in this module, as a kind of "Option".

Reply:
This file was essentially the only file in the repo when we started.  @samuelgruetter created it, so he may be better able to answer questions about it.  Of course, we can take this as an opportunity to clean it up.

Reply:
When I wrote this, I didn't call the module `Options`, but `Std`, with the intention that `Std` would contain the most common "very basic" definitions, and that you could get all of them with one single import.

Reply:
@samuelgruetter Sorry I was unclear; I meant that comment to apply to all of the questions about this file (see below).  In terms of naming, we could go back to `Std`, but I fear that doesn't communicate much to the reader; i.e., everyone who sees that name will need to open it up to see what's inside. 

@robin-aws We could split `Result` into it's own module, but that starts to seem like we're slicing things pretty thinly.  

Is there another name that would better connote `Option` + `Result`?

Reply:
I want to say `Monads` would make sense, except that I've been explicitly avoiding making that connection since `:-` is explicitly NOT a general-purpose monadic operator, and I'm pessimistic about monadic syntax playing nicely with a language with explicit imperative side-effects (i.e. being able to mutate state with regular statements). We had a lot of discussion about that in https://github.com/dafny-lang/dafny/pull/455

I'm almost happier with `Results` since `Option` is often used to model an optional result, although that's selling `Option` short a bit since it's also useful for optional input.